core/src/test/java/org/apache/iceberg/TestPartitionSpecInfo.java

```diff
@@ -95,6 +96,30 @@ public void testSpecInfoPartitionedTable() {
         .doesNotContainKey(Integer.MAX_VALUE);
   }
 
+  @TestTemplate
+  public void testSpecInfoPartitionedTableCaseInsensitive() {
+    PartitionSpec spec =
+        PartitionSpec.builderFor(schema).caseSensitive(false).identity("DATA").build();
+    TestTables.TestTable table = TestTables.create(tableDir, "test", schema, spec, formatVersion);
+
+    assertThat(table.spec()).isEqualTo(spec);
+    assertThat(table.spec().lastAssignedFieldId()).isEqualTo(spec.lastAssignedFieldId());
+    assertThat(table.specs())
+        .containsExactly(entry(spec.specId(), spec))
+        .doesNotContainKey(Integer.MAX_VALUE);
```

Comment:
for my education, what is this assertion for?

Reply:
Are you asking about the specific assert below?  Actually, it doesn't matter. All I did was take the test above, `testSpecInfoPartitionedTable` on line 88, and make a case-insensitive version of the test. To understand the purpose of the assert you'll have to ask the person who wrote `testSpecInfoPartitionedTable`.

```
assertThat(table.specs())
        .containsExactly(entry(spec.specId(), spec))
        .doesNotContainKey(Integer.MAX_VALUE);
```